.changeset/brave-seas-impress.md

```diff
@@ -0,0 +1,7 @@
+---
+"@xstate/cli": minor
+"@xstate/tools-shared": minor
+"stately-vscode": minor
+---
+
+Gave typegen the ability to calculate all of the events which can fire exit actions.
```

Comment:
At the moment this will not be type-safe at all. Do we want to wait for landing [the PR in the core](https://github.com/statelyai/xstate/pull/3126) before landing this PR here? I feel like probably a coordinated release would be better. With a callout that this will require a new version of XState to work properly. 

I will also be working next week on "predictable event types for implementations" and we could probably batch releasing it with this change here.

Reply:
Yes, I can dig that. It's a relatively small change this end so we should co-ordinate the release together.